symfit/core/printing.py

```diff
@@ -31,6 +31,32 @@ class SymfitNumPyPrinter(NumPyPrinter):
     Our own NumpyPrinter subclass, in case we need to print certain numpy
     features which are not yet supported in SymPy.
     """
+    def _print_MatMul(self, expr):
+        """
+        Matrix multiplication printer. The sympy one turns everything into a
+        dot product without type-checking.
+        """
+        from sympy import MatrixExpr
+        links = []
+        for i, j in zip(expr.args[1:], expr.args[:-1]):
+            if isinstance(i, MatrixExpr) and isinstance(j, MatrixExpr):
+                links.append(').dot(')
+            else:
+                links.append('*')
+        printouts = [self._print(i) for i in expr.args]
+        result = [printouts[0]]
+        for link, printout in zip(links, printouts[1:]):
+            result.extend([link, printout])
+        return '({0})'.format(''.join(result))
+
+    def _print_Inverse(self, expr):
+        return "%s(%s)" % (self._module_format('numpy.linalg.inv'),
+                           self._print(expr.args[0]))
+
+    def _print_HadamardProduct(self, expr):
```

Comment:
I'm guessing this is the element wise product?

Reply:
Yeah, that's the formal mathematical name for it, and how it's called in sympy.